Improve the migration guide (#5430)
containous/traefik: docs/content/migration/v1-to-v2.md

        "test:$apr1$H6uskkkW$IgXLP6ewTrSuBkTrqE8wj/"
      ]
    ```
    
    ```yaml tab="File (YAML)"
    ## static configuration
    # traefik.yaml
    
    entryPoints:
      web-secure:
        address: ':443'
        
    api: {}
    
    providers:
      file:
        filename: /dymanic-conf.yaml
   
    ##---------------------##
    
    ## dynamic configuration
    # dymanic-conf.yaml
    
     http:
      routers:
        api:
          rule: Host(`traefik.docker.localhost`)
          entrypoints:
            - web-secure
          service: api@internal
          middlewares:
            - myAuth
          tls: {}
          
      middlewares:
        myAuth:
          basicAuth:
            users:
              - 'test:$apr1$H6uskkkW$IgXLP6ewTrSuBkTrqE8wj/'
    ``` 
    
## Providers

Supported [providers](../providers/overview.md), for now:

* [ ] Azure Service Fabric
* [ ] BoltDB
* [ ] Consul
* [ ] Consul Catalog
* [x] Docker
* [ ] DynamoDB
* [ ] ECS
* [ ] Etcd
* [ ] Eureka
* [x] File
* [x] Kubernetes Ingress (without annotations)
* [x] Kubernetes IngressRoute
* [x] Marathon
* [ ] Mesos
* [x] Rancher
* [x] Rest
* [ ] Zookeeper

## Some Tips You Should Known

> * The different sources of static configuration (file, CLI flags, ...) cannot be [mixed](../getting-started/configuration-overview.md#the-static-configuration).

Reviewer: ```suggestion
* Different sources of static configuration (file, CLI flags, ...) cannot be [mixed](../getting-started/configuration-overview.md#the-static-configuration).
```

> * Now we can refers to configuration element between different providers by using the provider namespace (`@provider`).

Reviewer: ```suggestion
* Now, configuration elements can be referenced between different providers by using the namespace notation: `@<provider>`.
```

>   For example a router in a File Provider can refer to a service defined in Docker Provider.

Reviewer: ```suggestion
  For instance, a router named `webapp` in a File Provider can refer to a service named `backoffice` defined in Docker Provider with the following notation: `service=backoffice@docker`.
```

> * The middlewares are applied in the same order as their declaration in router.

Reviewer: ```suggestion
* Middlewares are applied in the same order as their declaration in router.
```
